File: ruby/test/test_embed_users.rb
# frozen_string_literal: true

require_relative './base_template'

SAMPLE_EMBED_USER_PARAMS = {
  id: 'sample-embed-user-id',
  name: 'Rails Integration Test User',
  email: 'test@test.com',
  metadata: {
    hello: 'world'
  }
}.freeze

# All tests related to Onvo's embed user endpoints
class EmbedUsersTest < BaseTemplate
  def before_all
    super
    @sample_embed_user_id = @onvo.embed_users.upsert(
      SAMPLE_EMBED_USER_PARAMS
    )['id'] # skipping test create
  end

  def test_list_embed_users
    assert_silent { @onvo.embed_users.list }
  end

  def test_get_embed_user
    assert_silent { @onvo.embed_users.get(@sample_embed_user_id) }
  end

  def test_get_embed_user_access_token
    assert_silent { @onvo.embed_users.get_access_token(@sample_embed_user_id) }
  end

  def after_all
    @onvo.embed_users.delete(@sample_embed_user_id) # skipping test delete
  end
end

File: python/tests/test_embed_users.py
from .base_test import BaseTest

SAMPLE_EMBED_USER_PARAMS = {
    "id": "sample-embed-user-id",
    "name": "Python Integration Test User",
    "email": "test@test.com",
    "metadata": {"hello": "world"},
}


class TestEmbedUsers(BaseTest):

    @classmethod
    def setUpClass(cls):
        super().setUpClass()
        cls.sampleEmbedUserId = cls.onvoSDK.embed_users.upsert(
            SAMPLE_EMBED_USER_PARAMS
        )[
            "id"
        ]  # Skipping test create

    def test_list(self):
        self.assertShouldRaise(None, self.onvoSDK.embed_users.list)

    def test_get(self):
        self.assertShouldRaise(
            None, lambda: self.onvoSDK.embed_users.get(self.sampleEmbedUserId)
        )

    def test_get_access_token(self):
        self.assertShouldRaise(
            None,
            lambda: self.onvoSDK.embed_users.get_access_token(self.sampleEmbedUserId),
        )

    def test_upsert_update(self):
        self.assertShouldRaise(
            None,
            lambda: self.onvoSDK.embed_users.upsert(
                {"id": self.sampleEmbedUserId, "name": "Renamed Python Test User"},
            ),
        )

    @classmethod
    def tearDownClass(cls) -> None:
        cls.onvoSDK.embed_users.delete(cls.sampleEmbedUserId)  # Skipping test delete
        super().tearDownClass()
